moveit_core/collision_detection_bullet/CMakeLists.txt

```diff
@@ -17,7 +17,7 @@ add_dependencies(${MOVEIT_LIB_NAME} ${catkin_EXPORTED_TARGETS})
 
 add_library(collision_detector_bullet_plugin src/collision_detector_bullet_plugin_loader.cpp)
 set_target_properties(collision_detector_bullet_plugin PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-target_link_libraries(collision_detector_bullet_plugin ${catkin_LIBRARIES} ${MOVEIT_LIB_NAME})
+target_link_libraries(collision_detector_bullet_plugin ${catkin_LIBRARIES} ${MOVEIT_LIB_NAME} moveit_planning_scene)
```

Comment:
Shouldn't `moveit_planning_scene` come from `${catkin_LIBRARIES}`?  Why does it need to be manually added here?

Reply:
This is the error on macOS if we don't add moveit_planning_scene:
```
 Undefined symbols for architecture x86_64:
  "planning_scene::PlanningScene::setActiveCollisionDetector(std::__1::shared_ptr<collision_detection::CollisionDetectorAllocator> const&, bool)", referenced from:
      collision_detection::CollisionDetectorFCLPluginLoader::initialize(std::__1::shared_ptr<planning_scene::PlanningScene> const&, bool) const in collision_detector_fcl_plugin_loader.cpp.o
ld: symbol(s) not found for architecture x86_64
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
```
(see https://github.com/RoboStack/moveit/runs/2347862944?check_suite_focus=true)

Reply:
I'm not sure why it should come from `${catkin_LIBRARIES}`. It's an internal dependency, and my bet is that it just happens that it's not built in the "right" order in some cases. Adding the dependency makes it explicit that `moveit_planning_scene` needs to be built first.